crates/hyperdrive-math/src/long/max.rs

```diff
@@ -81,6 +81,12 @@ impl State {
         // we converge to the solution.
         let mut max_base_amount =
             self.max_long_guess(absolute_max_base_amount, checkpoint_exposure);
+
+        // possible_max_base_amount might be less than minimum transaction amount.
+        // we clamp here if so
+        if max_base_amount < self.minimum_transaction_amount() {
```

Comment:
We might want to return 0 if we get here. If the max long is less than the minimum transaction amount, then the long can’t be opened. Ofc, we can do this at the end of the function instead of short-circuiting Newton’s. Nice catch!

Reply:
Makes sense. I'll keep this check here just in case `max_long_guess` underestimates too much, and allow newton's method to iterate using this as an initial guess.